src/services/ui/src/components/Opensearch/Filtering/useFilterDrawer.ts

```diff
@@ -0,0 +1,92 @@
+import { useState, useEffect } from "react";
```

Comment:
Main functional focus of the filter drawer is to capture local state changes when the drawer is open.  
Then commit state changes to the url hash after the drawer collapses.  
This is to avoid any micro filter changes from preemptively triggering os requests.